src/UI/Components/Diagram/actions.ts

```diff
@@ -300,90 +316,33 @@ export function appendEmbeddedEntity(
   paper: dia.Paper,
   graph: dia.Graph,
   embeddedEntity: EmbeddedEntity,
-  entityAttributes: InstanceAttributeModel,
+  entityAttributes: InstanceAttributeModel | InstanceAttributeModel[],
   embeddedTo: string | null,
   holderName: string,
   instanceToConnectRelation?: ServiceEntityBlock,
   presentedAttr?: "candidate" | "active",
   isBlockedFromEditing?: boolean,
 ): ServiceEntityBlock[] {
   //Create shape for Entity
-  const flatAttributes = embeddedEntity.attributes.map(
-    (attribute) => attribute.name,
-  );
 
-  if (Array.isArray(entityAttributes)) {
-    const createdInstances: ServiceEntityBlock[] = [];
-
-    (entityAttributes as InstanceAttributeModel[]).map((entityInstance) => {
-      const instanceAsTable = new ServiceEntityBlock()
-        .setTabColor("embedded")
-        .setName(embeddedEntity.name);
-
-      appendColumns(instanceAsTable, flatAttributes, entityInstance);
-      instanceAsTable.set("isEmbedded", true);
-      instanceAsTable.set("holderName", holderName);
-      instanceAsTable.set("embeddedTo", embeddedTo);
-      instanceAsTable.set("isInEditMode", true);
-      instanceAsTable.set("isBlockedFromEditing", isBlockedFromEditing);
-      //add to graph
-      instanceAsTable.addTo(graph);
-
-      createdInstances.push(instanceAsTable);
-      //iterate through embedded entities to create and connect them
-      embeddedEntity.embedded_entities.forEach((entity) => {
-        //we are basing iteration on service Model, if there is no value in the instance, skip that part
-        if (!entityInstance[entity.name]) {
-          return;
-        }
-        const appendedEntities = appendEmbeddedEntity(
-          paper,
-          graph,
-          entity,
-          entityInstance[entity.name] as InstanceAttributeModel,
-          instanceAsTable.id as string,
-          embeddedEntity.name,
-          instanceToConnectRelation,
-          presentedAttr,
-          isBlockedFromEditing,
-        );
-        appendedEntities.map((appendedEntities) => {
-          handleInfoIcon(appendedEntities, presentedAttr);
-        });
-        connectEntities(
-          graph,
-          instanceAsTable,
-          appendedEntities,
-          isBlockedFromEditing,
-        );
-      });
-
-      embeddedEntity.inter_service_relations?.map((relation) => {
-        const relationId = entityAttributes[relation.name] as relationId;
-        if (relationId) {
-          instanceAsTable.addRelation(relationId, relation.name);
-          if (
-            instanceToConnectRelation &&
-            relationId === instanceToConnectRelation.id
-          ) {
-            connectEntities(
-              graph,
-              instanceAsTable,
-              [instanceToConnectRelation],
-              isBlockedFromEditing,
-            );
-          }
-        }
-      });
-    });
+  /**
+   *  Extracted logic append Embedded entity to reduce repeats of code
```

Comment:
I don't think this is a good description honestly... the function is called handleSingleEntity, but the description says it's appending an Embedded entity 🤔

Reply:
still waiting for a better description :)